bin/oref0-autotune.py

```diff
@@ -161,50 +152,50 @@ def get_nightscout_bg_entries(nightscout_host, start_date, end_date, directory):
         res = requests.get(url)
         with open(os.path.join(directory, 'autotune', 'ns-entries.{date}.json'.format(date=date.strftime("%Y-%m-%d"))), 'w') as f:
-            f.write(res.text.encode('utf-8'))
+            f.write(six.ensure_str(res.text, 'utf-8'))
 
 def run_autotune(start_date, end_date, number_of_runs, directory):
     date_list = [start_date + datetime.timedelta(days=x) for x in range(0, (end_date - start_date).days)]
     autotune_directory = os.path.join(directory, 'autotune')
+    FNULL = open(os.devnull, 'w')
     for run_number in range(1, number_of_runs + 1):
         for date in date_list:
             # cp profile.json profile.$run_number.$i.json
             shutil.copy(os.path.join(autotune_directory, 'profile.json'),
                         os.path.join(autotune_directory, 'profile.{run_number}.{date}.json'
                         .format(run_number=run_number, date=date.strftime("%Y-%m-%d"))))
-        
-            # Autotune Prep (required args, <pumphistory.json> <profile.json> <glucose.json>), output prepped glucose 
+
+            # Autotune Prep (required args, <pumphistory.json> <profile.json> <glucose.json>), output prepped glucose
             # data or <autotune/glucose.json> below
             # oref0-autotune-prep ns-treatments.json profile.json ns-entries.$DATE.json > autotune.$RUN_NUMBER.$DATE.json
             ns_treatments = os.path.join(autotune_directory, 'ns-treatments.json')
             profile = os.path.join(autotune_directory, 'profile.json')
+            pump_profile = os.path.join(autotune_directory, "pumpprofile.json")
             ns_entries = os.path.join(autotune_directory, 'ns-entries.{date}.json'.format(date=date.strftime("%Y-%m-%d")))
-            autotune_prep = 'oref0-autotune-prep {ns_treatments} {profile} {ns_entries}'.format(ns_treatments=ns_treatments, profile=profile, ns_entries=ns_entries)
-            
-            # autotune.$RUN_NUMBER.$DATE.json  
+
+            # autotune.$RUN_NUMBER.$DATE.json
             autotune_run_filename = os.path.join(autotune_directory, 'autotune.{run_number}.{date}.json'
                                                  .format(run_number=run_number, date=date.strftime("%Y-%m-%d")))
-            with open(autotune_run_filename, "w+") as output:
-                logging.info('Running {script}'.format(script=autotune_prep))
-                call(autotune_prep, stdout=output, shell=True)
-                logging.info('Writing output to {filename}'.format(filename=autotune_run_filename))
-        
-            # Autotune  (required args, <autotune/glucose.json> <autotune/autotune.json> <settings/profile.json>), 
+            autotune_prep = 'oref0-autotune-prep {ns_treatments} {profile} {ns_entries} {pump_profile} --output-file {autotune_run_filename}'.format(ns_treatments=ns_treatments, profile=profile, ns_entries=ns_entries, pump_profile=pump_profile, autotune_run_filename=autotune_run_filename)
```

Comment:
`pumpprofile.json` is a required argument to `autotune-prep`. Though FWICT it's the same as `profile.json` so I'm not sure...why it's required here.

Reply:
The pumpprofile.json represents the pump's settings. Once autotune has been run, the profile.json should get updated to reflect the autotuned settings, but the pumpprofile.json should keep the pump's settings in order to anchor the range over which autotune can safely adjust settings.